Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -119,4 +124,52 @@ async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingProfile,
     }
 }
 
+/**
+ * Executes Web Deploy command
+ * 
+ * @param   webDeployPkg                   Web deploy package
+ * @param   webAppName                      web App Name
+ * @param   publishingProfile               Azure RM Connection Details
+ * @param   removeAdditionalFilesFlag       Flag to set DoNotDeleteRule rule
+ * @param   excludeFilesFromAppDataFlag     Flag to prevent App Data from publishing
+ * @param   takeAppOfflineFlag              Flag to enable AppOffline rule
+ * @param   virtualApplication              Virtual Application Name
+ * @param   setParametersFile               Set Parameter File path
+ * @param   additionalArguments             Arguments provided by user
+ * 
+ */
+async function DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishingProfile, virtualApplication, isFolderBasedDeployment) {
+
+    var isDeploymentSuccess = true;
+    var deploymentError = null;
+    try {
+        
+        var virtualApplicationMappings = azureWebAppDetails.properties.virtualApplications;
+        var webAppZipFile = webDeployPkg;
+        if(isFolderBasedDeployment){
+            webAppZipFile = tl.getVariable('System.DefaultWorkingDirectory') + randomstring.generate(7) + '.zip';
```

Comment:
Yes, may Vincent point is right that way we can avoid dependency on randomstring generator one less OSS component. 

Because at any point of time there can be only one task can be run by the agent and hence no duplicate name issues. 

@kmkumaran  Your thoughts?